c/validation/adbc_validation.cc

```diff
@@ -430,6 +430,20 @@ void ConnectionTest::TestMetadataGetTableSchema() {
                                     {"strings", NANOARROW_TYPE_STRING, NULLABLE}}));
 }
 
+void ConnectionTest::TestMetadataGetTableSchemaEscaping() {
+  if (!quirks()->supports_bulk_ingest(ADBC_INGEST_OPTION_MODE_CREATE)) {
+    GTEST_SKIP();
+  }
+  ASSERT_THAT(AdbcConnectionNew(&connection, &error), IsOkStatus(&error));
+  ASSERT_THAT(AdbcConnectionInit(&connection, &database, &error), IsOkStatus(&error));
+
+  Handle<ArrowSchema> schema;
+  ASSERT_NE(AdbcConnectionGetTableSchema(&connection, /*catalog=*/nullptr,
+                                         /*db_schema=*/nullptr, "(SELECT CURRENT_TIME)",
+                                         &schema.value, &error),
+            ADBC_STATUS_OK);
```

Comment:
Somewhat unfortunate that postgres returns an `INVALID_ARGUMENT` error code whereas sqlite returns `NOT_FOUND` . I didn't see much in the sqlite error handling docs that would allow us to differentiate in that driver

Reply:
I thought I fixed this in the last commit

Reply:
Ah, you want INVALID_ARGUMENT specifically this time...

Reply:
I'm not sure which one is better. Maybe should return a NOT_FOUND since that is the least common denominator across both drivers?

Reply:
Should clarify though that the Postgres error is thrown from the PqResultHelper class not from GetTableSchema directly

Reply:
I think NOT_FOUND makes more sense. In Postgres you can check the SQLSTATE, that's what I did when I fixed up something similar to this earlier